### PR TITLE
mixing: Unlock wallets first.

### LINKED ISSUE #2856: mixing: Starting with locked wallet causes problems.
Clicking the mix slider for dcr when the wallet is locked errors with:
```
2024-07-04 05:12:17.666 [DBG] CORE[dcr][SPV]: Starting cspp peer-to-peer funds mixer
2024-07-04 05:12:17.680 [ERR] CORE[dcr][SPV]: Error mixing account: wallet.MixAccount: wallet locked:
	wallet.MixOutput(1212c06d1ee9af17d4e8a09bfafa98e216da23d5eedb0271107f59b0d4344643:0): account with unique passphrase is locked
```

Restarting bisonw it will error again because the wallet is locked.

The locked wallet also shows this error when trying to turn off mixing:
```
2024-07-04 05:21:38.643 [TRC] CORE[dcr]: Obtained estimate for 2-conf fee rate, 10
2024-07-04 05:21:38.647 [ERR] CORE[dcr][SPV]: signature error for index 0: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.647 [ERR] CORE[dcr][SPV]: signature error for index 1: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.647 [ERR] CORE[dcr][SPV]: signature error for index 2: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.647 [ERR] CORE[dcr][SPV]: signature error for index 3: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.647 [ERR] CORE[dcr][SPV]: signature error for index 4: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.647 [ERR] CORE[dcr][SPV]: signature error for index 5: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.647 [ERR] CORE[dcr][SPV]: signature error for index 6: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.647 [ERR] CORE[dcr][SPV]: signature error for index 7: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.647 [ERR] CORE[dcr][SPV]: signature error for index 8: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.647 [ERR] CORE[dcr][SPV]: signature error for index 9: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
....
2024-07-04 05:21:38.648 [ERR] CORE[dcr][SPV]: signature error for index 108: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.648 [ERR] CORE[dcr][SPV]: signature error for index 109: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.648 [ERR] CORE[dcr][SPV]: signature error for index 110: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.648 [ERR] CORE[dcr][SPV]: signature error for index 111: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
2024-07-04 05:21:38.648 [ERR] CORE[dcr][SPV]: signature error for index 112: wallet.SignTransaction: wallet locked: account with unique passphrase is locked
```

I've also seen this when not messing with mixing.

### PR DESCRIPTION
**Pull request by JoeGruffins**
closes #2856

### CODE REVIEW

File: client/asset/dcr/native_wallet.go [hunk 1]
```diff
@@ -115,18 +115,11 @@ func (w *NativeWallet) Connect(ctx context.Context) (*sync.WaitGroup, error) {
 	if err != nil {
 		return nil, err
 	}
-	if w.mixing.Load() {
-		w.startFundsMixer()
-	} else {
-		// Ensure any funds in the mixed account are transferred back to
-		// primary.
-		w.stopFundsMixer()
```
Comment (JoeGruffins) [inline client/asset/dcr/native_wallet.go:123]: This also did not return the error from `w.stopFundsMixer()` which is a bug.

File: client/core/core.go [hunk 2]
```diff
@@ -4658,6 +4658,24 @@ func (c *Core) connectWallets() {
...
 			}
+			if mw, is := wallet.Wallet.(asset.FundsMixer); is {
+				startMixing := func() error {
+					stats, err := mw.FundsMixingStats()
+					if err != nil {
+						return fmt.Errorf("error checking %s wallet mixing stats: %v", unbip(wallet.AssetID), err)
+					}
+					if err := c.connectAndUnlock(crypter, wallet); err != nil {
+						return fmt.Errorf("error unlocking %s wallet for mixing: %v", unbip(wallet.AssetID), err)
+					}
```
Comment (martonp) [inline client/core/core.go:4669]: Don't need to unlock if `!stats.Enabled`

**Review by martonp — APPROVED** [review]
LGTM. Should we block users from locking the wallet if mixing is running though?

**Comment by JoeGruffins** [comment]
> Should we block users from locking the wallet if mixing is running though?

It looks like locking should already be stopped in dcr.go but I'm unsure if it's working...
https://github.com/decred/dcrdex/blob/f41c91ed8cef26b6fcceacc786f0115abaad6c79/client/asset/dcr/dcr.go#L4165-L4167

File: client/core/core.go (hunk 2 above)
Comment (JoeGruffins) [reply client/core/core.go:4669]: Even disabled can move funds if any are still left in mixed and unmixed accounts. The stats could include this info however...

**Comment by JoeGruffins** [comment]
Preventing lock when mixing and not unlocking if there are not funds to move and mixing not enabled https://github.com/decred/dcrdex/compare/f776e7f25766b024b7dae50a36100229e751f283..1d2bca1f747b559b2e3da17e79a532acc8fae659

**Review by martonp — APPROVED** [review]

**Pull request merged**